### DISCUSSION TITLE
DNS over Tor

### DISCUSSION THREAD

**Discussion by gavinowens** [0xERR0R/blocky#430] [Q&A] [answered]
Do you have any plans to implement DNS over Tor? I know PiHole has this, but I was wondering if you did.

**Comment by 0xERR0R** [comment, marked as answer]
I have currently no plans to implement it.

**Reply by gavinowens** [reply]
> I have currently no plans to implement it.

Ok, thanks!

**Comment by kwitsch** [comment]
@awholegnuworld I'm curious on what you are referring.

If It's the basic setup from the [documentation](https://docs.pi-hole.net/guides/misc/tor/setup/) should be possible by default.
Simply follow the first steps and use "127.0.10.1" as upstream resolver.

If you like to resolve requests over TLS or HTTPS through Tor its a bit more complicated.
Easiest way i can think of would be setting up Tor with an additional HTTPS proxy and running blocky in docker with the corresponding proxy option set.

**Reply by gavinowens** [reply]
Yeah, I was talking about TLS over Tor. Thanks for the input though!

**Reply by kwitsch** [reply]
A generic feature request would be the implementation of an SOCKS proxy configuration option in blocky.
Since Tor itself is just a SOCKS proxy to other programs and the useacase wouldn't be limited to it.

Took a short look into it and it seems that it could be done with a feasible amount of work since there is a [SOCKS5 function](https://pkg.go.dev/golang.org/x/net/proxy#SOCKS5) in net/proxy.
